libp2p/protocols/pubsub/pubsub.nim

```diff
@@ -60,48 +61,24 @@ type
     msgIdProvider*: MsgIdProvider      # Turn message into message id (not nil)
     msgSeqno*: uint64
 
-method handleDisconnect*(p: PubSub, peer: PubSubPeer) {.base.} =
+method unsubscribePeer*(p: PubSub, peer: PubSubPeer) {.base.} =
   ## handle peer disconnects
   ##
-  
-  if not(isNil(peer)) and peer.peerInfo notin p.conns:
-    trace "deleting peer", peer = peer.id
-    peer.onConnect.fire() # Make sure all pending sends are unblocked
-    p.peers.del(peer.id)
-    trace "peer disconnected", peer = peer.id
 
-    # metrics
-    libp2p_pubsub_peers.set(p.peers.len.int64)
-
-proc onConnClose(p: PubSub, conn: Connection) {.async.} =
-  try:
-    let peer = conn.peerInfo
-    await conn.closeEvent.wait()
-
-    if peer in p.conns:
-      p.conns[peer].excl(conn)
-      if p.conns[peer].len <= 0:
-        p.conns.del(peer)
-
-    if peer.id in p.peers:
-      p.handleDisconnect(p.peers[peer.id])
-
-  except CancelledError as exc:
-    raise exc
-  except CatchableError as exc:
-    trace "exception in onConnClose handler", exc = exc.msg
+  peer.subscribed = false
+  libp2p_pubsub_peers.set(p.peers.len.int64)
 
 proc sendSubs*(p: PubSub,
                peer: PubSubPeer,
                topics: seq[string],
                subscribe: bool) {.async.} =
   ## send subscriptions to remote peer
-  asyncCheck peer.sendSubOpts(topics, subscribe)
+  await peer.sendSubOpts(topics, subscribe)
```

Comment:
remove `await` and `async` - one future less

Reply:
sending is `async`